proto/README.md

```diff
@@ -0,0 +1,62 @@
+The protos within this directory hierarchy represent the data within
+events generated by Google eventing infrastructure, usually
+represented as [CloudEvents](https://cloudevents.io/). They are
+provided as a machine-readable spec, allowing libraries of data
+types to be created to make it easier to consume events.
+
+There are two messages that are relevant for any given CloudEvent
+type:
+
+- The "envelope" message (e.g. `ObjectFinalized`)
+- The "data" message (e.g. `StorageObject`)
+
+There is a 1:1 relationship between CloudEvent types and envelope
+messages, but a many:1 relationship between CloudEvent types and
+data messages. In other words, many CloudEvent types can use the
+same data message. For example, every event associated with
+something happening to a Cloud Storage object uses `StorageObject`
+as the data message. Each envelope message specifies the CloudEvent
+type it's associated with via a proto annotation.
+
+Note that each proto package is versioned, but this version is
+independent of any API version. It would be possible for a single
+API version to support multiple event versions, or vice versa.
+
+Our convention for event types is to have a dot-separated list of:
+
+- `google` (always)
```

Comment:
I think this README could use more visually appealing and structured information such as this table rather than large sections of text and list.

Maybe we can get a 3rd opinion?